### PR TITLE
Channel-specific review groups

### PR DESCRIPTION
**Pull request by iamvery**
The benefits I see are:
* No change to the current interface
* You ​_have_​ to request assignment in the appropriate channel which may lead to collaboration from others*

*@kylie mentioned this to me as a benefit. I didn’t recognize it at first, but I really like the idea now. It sort of forces review assignments to be part of the conversation which is probably a win. Also, for what it's worth that's by in large how we've done things anyway.

cc @kyfast @tdouce

### CODE REVIEW

**Comment by wolfbrain** [comment]
:eyes: @sebastiangeiger

File: lib/lita/handlers/reviewme.rb [hunk 1]
```diff
@@ -96,16 +96,16 @@ def comment_on_github(response)
...
+        reviewer = next_reviewer(room)
         response.reply(chat_mention(reviewer, url))
       end
 
       private
 
-      def next_reviewer
-        redis.rpoplpush(REDIS_LIST, REDIS_LIST)
+      def next_reviewer(room)
+        ns_redis(room).rpoplpush(REDIS_LIST, REDIS_LIST)
```
Comment (sebastiangeiger) [inline lib/lita/handlers/reviewme.rb]: Am I correct to assume, that `ns_redis(room)` would serialize the room to redis and then use it as a key? Could you instead use a shorter key like `"room:#{room.id}". (That assumes that `room` has a unique id)

Comment (iamvery) [inline lib/lita/handlers/reviewme.rb]: So I found out that all redis connections within a plugin are already namespaced. So `redis` in this case would be something like `lita:reviewme` as the key. This just adds another bit to that so you end up with something like `lita:reviewme:R12345` (slack channel id). Does that make sense?

Comment (sebastiangeiger) [inline lib/lita/handlers/reviewme.rb]: That makes sense, I was afraid that room is an object and it would serialize the full object.

Comment (iamvery) [inline lib/lita/handlers/reviewme.rb]: I think that's actually something silly like `room_object` https://github.com/litaio/lita/blob/94ad234c9737d658d1b0d8d7829530e4b378dc62/lib/lita/message.rb#L30

File: lib/lita/handlers/reviewme.rb [hunk 2]
```diff
@@ -123,6 +123,14 @@ def github_access_token
       def chat_mention(reviewer, url)
         "#{reviewer}: :eyes: #{url}"
       end
+
+      def get_room(response)
+        response.message.source.room
```
Comment (iamvery) [inline lib/lita/handlers/reviewme.rb]: Hmm, room appears to be delegated, so I can violate demeter slightly less by removing `.source` here https://github.com/litaio/lita/blob/94ad234c9737d658d1b0d8d7829530e4b378dc62/lib/lita/message.rb#L30

Comment (iamvery) [inline lib/lita/handlers/reviewme.rb]: It looks like `room_object` is delegated to the source (https://github.com/litaio/lita/blob/94ad234c9737d658d1b0d8d7829530e4b378dc62/lib/lita/message.rb#L30). I wonder if it'd be _slightly better_ do so something like.

```ruby
response.message.room_object

# and elsewhere
room.name
```

**Comment by stevenharman** [comment]
You might also want to mention this behavior in the `README`. I could be quite surprising (even though it's also quite desirable) to have this behavior scoped to a channel.

**Comment by iamvery** [comment]
Great point! We've been chatting up a couple other small changes. I'll make sure this is noted. Thanks!

**Pull request merged**